docs/api-reference/next/router.md

```diff
@@ -112,6 +112,8 @@ export default function Page() {
 }
 ```
 
+> **Note:** When navigating to the same page in Next.js, the page's state **will not** be reset by default, as the top-level React component is the same. You can manually ensure the state is updated using `useEffect` or by adding a `key` prop with the `asPath` to the components who's state should be reset.
```

Comment:
```suggestion
> **Note:** When navigating to the same page in Next.js, the page's state **will not** be reset by default, as the top-level React component is the same. You can manually ensure the state is updated using `useEffect`.
```

@leerob we can remove this bit if that sounds good?

Reply:
Sounds great!